### DISCUSSION TITLE
How to get the current logged in user

### DISCUSSION THREAD

**Discussion by Sourav-techlur** [sl-nx/superlogin-next#53] [General]
Hi,

The document suggest that /session is deprecated but what should be the alternate of this to get the logged in user information?

Any example will be appreciated.

Thank you

**Comment by fynnlyte** [comment]
Currently, a session is valid if there is a corresponding entry in `_users`. The `/session` - route simply checks whether the doc exists and the `password` matches. So making an authenticated request to the CouchDB root url (assuming that you have `require_valid_user` switchedq on) or to a user's personal DB is equivalent to `/session`.

But i should probably remove the deprecation. I'm considering to re-introduce `redis` as optional dependency since it's easier to invalidate sessions that way (`couch-auth` requires you to run `removeExpiredKeys()` regularily) and more performant for APIs.

**Reply by fynnlyte** [reply]
Also, `couchAuth.confirmSession(token, password)` will return the expiration date and the user's UUID if you want to integrate this elsewhere in your API

**Reply by Sourav-techlur** [reply]
Thank you @LyteFM `confirmSession` works fine.

**Comment by Sourav-techlur** [comment]
Hello @LyteFM 

Can you suggest me how can we share the database between two users. In our application we are providing one user access to another user. I read the documentation provided by you to update the _security rules or adding some configuration but i couldn't understand it properly. Can you explain it little bit in details.

Thank you

**Reply by fynnlyte** [reply]
You could have a look a the config for `shared` databases and the corresponding option in `addUserDB()`. But I guess that will only only work for databases that are shared across _all_ users.

I guess you'll need to develop your own logic for sharing the access between users. Have a look at https://docs.couchdb.org/en/stable/api/database/security.html and at the documents that `couch-auth` creates in `_users`. You'll always have the `key` of the current user as in `sl-users` inside the `roles` and the uuid in the doc itself, e.g.

```
// org.couchdb.user:mLDKyPHaTCCa_2ZcKHytOk has:
  "user_uid": "83bb2084-d4f6-4cc9-9d8d-a95645904e3b",
  "user_id": "ygkrk4wh",
  "roles": [
    "user:ygkrk4wh",
    "user"
  ],

// doc 83bb2084d4f64cc99d8da95645904e3b in sl-users has:
 "key": "ygkrk4wh",
```

So by adding `user:ygkrk4wh` to the member roles of a database, this user will have access to it. And you can use the `userCtx` to add some logic to the design docs, e.g. if you want to limit modifications to some documents.

Writing this comment, I realize that I should probably add `user:83bb2084d4f64cc99d8da95645904e3b` to the roles as well :D

**Reply by Sourav-techlur** [reply]
Hello @LyteFM 

Yes the user:key works fine actually i was trying with user_id and couch was giving 403.

Thanks for your help you have done an amazing job by creating this package.

**Reply by fynnlyte** [reply]
Probs for creating go to @colinskow but I've modernized and maintained this fork for  while now ;)
